src/kibana/components/vislib/visualizations/tile_map.js

```diff
@@ -208,36 +270,87 @@ define(function (require) {
+      var min = mapData.properties.allmin;
+      var max = mapData.properties.allmax;
 
-      // min and max from chart data for this map
-      // var min = mapData.properties.min;
-      // var max = mapData.properties.max;
+      var featureLayer = L.geoJson(mapData, {
+        pointToLayer: function (feature, latlng) {
+          var count = feature.properties.count;
+          var radius = self.geohashMinDistance(feature);
+          return L.circle(latlng, radius);
+        },
+        onEachFeature: function (feature, layer) {
+          self.bindPopup(feature, layer);
+        },
+        style: function (feature) {
+          var count = feature.properties.count;
+          var color = self.quantizeColorScale(count, min, max);
+          return {
+            fillColor: color,
+            color: self.darkerColor(color),
+            weight: 1.0,
+            opacity: 1,
+            fillOpacity: 0.75
+          };
+        }
+      });
+
+      // add legend
+      if (mapData.features.length > 1) {
+        self.addLegend(mapData, map);
+      }
+
+      return featureLayer;
+    };
+
+    /**
+     * Type of data overlay for map:
+     * creates featurelayer from mapData (geoJson)
+     * with rectangles that show the geohash grid bounds
+     *
+     * @method geohashGrid
+     * @param map {Object}
+     * @param mapData {Object}
+     * @return {undefined}
+     */
+    TileMap.prototype.shadedGeohashGrid = function (map, mapData) {
+      var self = this;
 
       // super min and max from all chart data
       var min = mapData.properties.allmin;
       var max = mapData.properties.allmax;
 
-      var length = mapData.properties.length;
-      var precision = mapData.properties.precision;
-      var zoomScale = self.zoomScale(mapZoom);
       var bounds;
-      var defaultColor = '#005baa';
+
       var featureLayer = L.geoJson(mapData, {
         pointToLayer: function (feature, latlng) {
           var count = feature.properties.count;
-          var rad = zoomScale * 3;
-          return L.circleMarker(latlng, {
-            radius: rad
-          });
+          var geohashRect = feature.properties.rectangle;
+          // get bounds from northEast[3] and southWest[1]
+          // points in geohash rectangle
+          var bounds = [
+            [geohashRect[3][1], geohashRect[3][0]],
+            [geohashRect[1][1], geohashRect[1][0]]
+          ];
+          return L.rectangle(bounds);
         },
         onEachFeature: function (feature, layer) {
           self.bindPopup(feature, layer);
+          layer.on({
+            mouseover: function (e) {
+              var layer = e.target;
+              // bring layer to front if not older browser
+              if (!L.Browser.ie && !L.Browser.opera) {
+                layer.bringToFront();
+              }
+            }
+          });
         },
         style: function (feature) {
           var count = feature.properties.count;
```

Comment:
This appears to be the same function as [https://github.com/elastic/kibana/pull/3445/files#diff-220771d5f556b3a25ecdffc1771b40afR292]().

Perhaps we can create one function to be shared between the two?